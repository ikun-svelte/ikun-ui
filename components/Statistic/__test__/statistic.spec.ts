import { afterEach, expect, test, describe, beforeEach } from 'vitest';
import KStatistic from '../src';
import KStatisticSuffix from './fixture/suffix.svelte';
import KStatisticPrefix from './fixture/prefix.svelte';
import { tick } from 'svelte';

let host;

const initHost = () => {
	host = globalThis.document.createElement('div');
	host.setAttribute('id', 'host');
	globalThis.document.body.appendChild(host);
};
beforeEach(() => {
	initHost();
});
afterEach(() => {
	host.remove();
});

describe('Test: KStatistic', () => {
	test('props: cls', async () => {
		// @ts-ignore
		const instance = new KStatistic({
			target: host,
			props: {
				cls: 'k-statistic--test'
			}
		});
		expect(instance).toBeTruthy();
		expect(host!.innerHTML.includes('k-statistic--test')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: title', async () => {
		// @ts-ignore
		const instance = new KStatistic({
			target: host,
			props: {
				title: 'test-title'
			}
		});
		expect(instance).toBeTruthy();
		expect(host.querySelector('.k-statistic__head')).toBeTruthy();
		expect(host!.innerHTML.includes('test-title')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: value', async () => {
		// @ts-ignore
		const instance = new KStatistic({
			target: host,
			props: {
				value: 57454157
			}
		});
		expect(instance).toBeTruthy();
		expect(host.querySelector('.k-statistic__content')).toBeTruthy();
		expect(host.querySelector('.k-statistic__value')).toBeTruthy();
		expect(host!.innerHTML.includes('57,454,157')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: precision', async () => {
		// @ts-ignore
		const instance = new KStatistic({
			target: host,
			props: {
				precision: 6,
				value: 268500.123456
			}
		});
		expect(instance).toBeTruthy();
		expect(host.querySelector('.k-statistic__content')).toBeTruthy();
		expect(host.querySelector('.k-statistic__value')).toBeTruthy();
		expect(host!.innerHTML.includes('268,500.123456')).toBeTruthy();
		// @ts-ignore
		instance.$set({ precision: 4 });
		await tick();
		expect(host!.innerHTML.includes('268,500.1234')).toBeTruthy();
	});

	test('slot: prefix', async () => {
		// @ts-ignore
		const instance = new KStatisticPrefix({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.querySelector('.k-statistic__content')).toBeTruthy();
		expect(host.querySelector('.k-statistic__value')).toBeTruthy();
		expect(host.querySelector('.k-statistic__prefix')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('slot: suffix', async () => {
		// @ts-ignore
		const instance = new KStatisticSuffix({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.querySelector('.k-statistic__content')).toBeTruthy();
		expect(host.querySelector('.k-statistic__value')).toBeTruthy();
		expect(host.querySelector('.k-statistic__suffix')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});
});
