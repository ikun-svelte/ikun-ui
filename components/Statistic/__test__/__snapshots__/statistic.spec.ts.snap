// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test: KStatistic > props: cls 1`] = `"<div class=\\"k-statistic k-statistic--test\\"> <div class=\\"k-statistic__content\\"> <span class=\\"k-statistic__value\\">0</span> </div></div>"`;

exports[`Test: KStatistic > props: title 1`] = `"<div class=\\"k-statistic\\"><div class=\\"k-statistic__head\\">test-title</div> <div class=\\"k-statistic__content\\"> <span class=\\"k-statistic__value\\">0</span> </div></div>"`;

exports[`Test: KStatistic > props: value 1`] = `"<div class=\\"k-statistic\\"> <div class=\\"k-statistic__content\\"> <span class=\\"k-statistic__value\\">57,454,157</span> </div></div>"`;

exports[`Test: KStatistic > slot: suffix 1`] = `"<div class=\\"k-statistic\\"><div class=\\"k-statistic__head\\">Daily active users</div> <div class=\\"k-statistic__content\\"> <span class=\\"k-statistic__value\\">268,500</span> <div class=\\"k-statistic__suffix\\"><span slot=\\"suffix\\">suffix</span></div></div></div>"`;
